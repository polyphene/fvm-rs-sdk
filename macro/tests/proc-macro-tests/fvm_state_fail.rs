#![allow(unreachable_code)]
use fvm_rs_sdk::*;

#[fvm_state]
pub struct MockStruct1<'a> {
    pub count: &'a u64,
}

#[fvm_state]
pub struct MockStruct2 {
    pub inner: InnerStruct,
}

struct InnerStruct {
    pub count: u64,
}

fn main() {}
