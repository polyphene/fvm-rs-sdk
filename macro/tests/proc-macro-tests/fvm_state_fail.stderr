error: structs with #[fvm_state] cannot have lifetime or type parameters
 --> tests/proc-macro-tests/fvm_state_fail.rs:5:23
  |
5 | pub struct MockStruct1<'a> {
  |                       ^^^^

error[E0277]: the trait bound `InnerStruct: fvm_rs_sdk::state::serde::Serialize` is not satisfied
   --> tests/proc-macro-tests/fvm_state_fail.rs:9:1
    |
9   | #[fvm_state]
    | ^^^^^^^^^^^^ the trait `fvm_rs_sdk::state::serde::Serialize` is not implemented for `InnerStruct`
    |
    = help: the following other types implement trait `fvm_rs_sdk::state::serde::Serialize`:
              &'a T
              &'a mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 201 others
    = note: required because of the requirements on the impl of `fvm_rs_sdk::state::serde::Serialize` for `&InnerStruct`
note: required by a bound in `serialize_newtype_struct`
   --> $CARGO/serde-1.0.140/src/ser/mod.rs
    |
    |         T: Serialize;
    |            ^^^^^^^^^ required by this bound in `serialize_newtype_struct`
    = note: this error originates in the derive macro `fvm_rs_sdk::encoding::tuple::Serialize_tuple` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `InnerStruct: fvm_rs_sdk::state::serde::Deserialize<'_>` is not satisfied
  --> tests/proc-macro-tests/fvm_state_fail.rs:11:16
   |
11 |     pub inner: InnerStruct,
   |                ^^^^^^^^^^^ the trait `fvm_rs_sdk::state::serde::Deserialize<'_>` is not implemented for `InnerStruct`
   |
   = help: the following other types implement trait `fvm_rs_sdk::state::serde::Deserialize<'de>`:
             &'a Path
             &'a [u8]
             &'a fvm_rs_sdk::fvm_ipld_encoding::cs_serde_bytes::Bytes
             &'a serde_bytes::bytes::Bytes
             &'a str
             ()
             (T0, T1)
             (T0, T1, T2)
           and 206 others

error[E0277]: the trait bound `InnerStruct: fvm_rs_sdk::state::serde::Deserialize<'_>` is not satisfied
    --> tests/proc-macro-tests/fvm_state_fail.rs:11:16
     |
11   |     pub inner: InnerStruct,
     |                ^^^^^^^^^^^ the trait `fvm_rs_sdk::state::serde::Deserialize<'_>` is not implemented for `InnerStruct`
     |
     = help: the following other types implement trait `fvm_rs_sdk::state::serde::Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a fvm_rs_sdk::fvm_ipld_encoding::cs_serde_bytes::Bytes
               &'a serde_bytes::bytes::Bytes
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
             and 206 others
note: required by a bound in `next_element`
    --> $CARGO/serde-1.0.140/src/de/mod.rs
     |
     |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `next_element`
